File: 백준/Silver/2559. 수열/수열.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
nums = list(map(int, input().split()))
intervalSumList = [sum(nums[:k])]

for i in range(n - k):
    intervalSum = intervalSumList[i] - nums[i] + nums[i+k]
    intervalSumList.append(intervalSum)

sys.stdout.write(str(max(intervalSumList)))

File: 백준/Silver/2559. 수열/수열.java
import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[] nums = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        ArrayList<Integer> sumList = new ArrayList<>();
        sumList.add(Arrays.stream(Arrays.copyOfRange(nums, 0, k)).sum());

        for (int i = 0; i < n - k; i++) {
            int intervalSum = sumList.get(i) - nums[i] + nums[i + k];
            sumList.add(intervalSum);
        }

        bw.write(Integer.toString(Collections.max(sumList)));
        bw.flush();

    }

}
